integration/tests/basic/rpc/test_rpc_base_calls.py

```diff
@@ -0,0 +1,650 @@
+            TestRpcBaseCalls._erc20_contract = contract, contract_deploy_tx, tx_receipt
+        return TestRpcBaseCalls._erc20_contract
+
+    def test_eth_call_without_params(self):
+        """Verify implemented rpc calls work eth_call without params"""
+        response = self.proxy_api.send_rpc("eth_call")
+        assert "error" in response, "Error not in response"
+
+    @pytest.mark.parametrize("tag", [Tag.LATEST, Tag.PENDING, Tag.EARLIEST])
+    def test_eth_call(self, tag):
+        """Verify implemented rpc calls work eth_call"""
+        params = [
+            {"to": self.recipient_account.address, "data": hex(pow(10, 14))},
+            tag.value,
+        ]
+
+        response = self.proxy_api.send_rpc("eth_call", params=params)
+        assert "error" not in response
+        assert (
+                response["result"] == "0x"
+        ), f"Invalid response result, `{response['result']}`"
+
+    def test_eth_get_transaction_receipt_with_incorrect_hash(self):
+        """Verify implemented rpc calls work eth_getTransactionReceipt when transaction hash is not correct"""
+
+        response = self.proxy_api.send_rpc(
+            method="eth_getTransactionReceipt", params=gen_hash_of_block(31)
+        )
+        assert "error" in response
+        assert response["error"]["message"] == "transaction-id is not hex"
+
+    def test_eth_gas_price(self):
+        """Verify implemented rpc calls work eth_gasPrice"""
+        response = self.proxy_api.send_rpc("eth_gasPrice")
+        assert "error" not in response
+        assert rpc_checks.is_hex(
+            response["result"]
+        ), f"Invalid current gas price `{response['result']}` in wei"
+
+    @pytest.mark.parametrize("param", [Tag.LATEST, Tag.PENDING, Tag.EARLIEST, None])
+    @pytest.mark.only_stands
+    def test_eth_get_balance(self, param: tp.Union[Tag, None]):
+        """Verify implemented rpc calls work eth_getBalance"""
+        response = self.proxy_api.send_rpc(
+            "eth_getBalance",
+            params=[self.sender_account.address, param.value if param else param],
+        )
+        if not param:
+            assert "error" in response, "Error not in response"
+            return
+        assert "error" not in response
+        assert rpc_checks.is_hex(response["result"]), AssertMessage.WRONG_AMOUNT.value
+
+    @pytest.mark.parametrize("param", [Tag.LATEST, Tag.PENDING, Tag.EARLIEST, None])
+    def test_eth_get_code(self, event_caller_contract, param: tp.Union[Tag, None]):
+        """Verify implemented rpc calls work eth_getCode"""
+        response = self.proxy_api.send_rpc(
+            "eth_getCode",
+            params=[event_caller_contract.address, param.value] if param else param,
+        )
+        if not param:
+            assert "error" in response, "Error not in response"
+            return
+        assert "error" not in response
+        assert (
+            is_hex(response["result"])
```

Comment:
Check length too

Reply:
Added length check and validation for value, that it's not consist only of zeros